cli/commands/list.ts

```diff
@@ -0,0 +1,45 @@
+import { findAllWorkspaceTasks } from "../workspace.ts";
+import { PrefixLogger } from "../prefix-logger.ts";
+import type { Task } from "../types.ts";
+
+export async function listCommand(rootCwd: string) {
+    // TODO: Include root tasks
+    const allWorkspaceTasks = await findAllWorkspaceTasks({ cwd: rootCwd });
+
+    if (allWorkspaceTasks.length === 0) {
+        console.log(`No tasks found`);
+        Deno.exit(1);
+    }
+
+    // Group package tasks by workspace
+    const groupedTasks = allWorkspaceTasks.reduce((acc, task) => {
+        if (!acc[task.package]) {
+            acc[task.package] = [];
+        }
+        acc[task.package].push(task);
+        return acc;
+    }, {} as Record<string, Task[]>);
+
+    // Calculate the max length of all task names
+    const taskNames: string[] = [];
+    Object.entries(groupedTasks).forEach(([_, workspaceTasks]) => {
+        workspaceTasks.forEach((task) => {
+            taskNames.push(task.task);
+        });
+    });
```

Comment:
suggestion (non-blocking): Consider using a `flatMap` off of `Object.entries(groupedTasks)` to accomplish the same thing.